db/dataset.py

```diff
@@ -183,7 +183,15 @@ def get_public_datasets():
             row = dict(row)
             dataset_id = row["id"].encode('utf-8')
             row["status"] = dataset_status[dataset_id]
-            datasets.append(row)
+            if(status_value == "all" or row["status"] == status_value):
```

Comment:
Parentheses here aren't necessary. You can try using [pylint](https://www.pylint.org/) which catches most of these problems.